pkg/controller/certificates/issuing/internal/secret.go

```diff
@@ -150,36 +152,46 @@ func (s *SecretsManager) setValues(crt *cmapi.Certificate, secret *corev1.Secret
 		secret.Data[cmmeta.TLSCAKey] = data.CA
 	}
 
+	if secret.Annotations == nil {
+		secret.Annotations = make(map[string]string)
+	}
+
+	if secret.Labels == nil {
+		secret.Labels = make(map[string]string)
+	}
+
+	if crt.Spec.SecretTemplate != nil {
+		for k, v := range crt.Spec.SecretTemplate.Labels {
+			secret.Labels[k] = v
+		}
+		for k, v := range crt.Spec.SecretTemplate.Annotations {
+			secret.Annotations[k] = v
+		}
+	}
+
 	var certificate *x509.Certificate
 	if len(data.Certificate) > 0 {
 		var err error
 		certificate, err = utilpki.DecodeX509CertificateBytes(data.Certificate)
-		// TODO: handle InvalidData here?
```

Comment:
I wonder what this TODO comment meant? 
Does it mean that cert-manager should tolerate invalid x.509 certificate bytes and add the annotations to the Secret anyway?

Reply:
Ah, I didn't understand, so I just removed this comment. What you are saying makes sense, I re-added the comment and added a bit more info.